x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -156,7 +156,7 @@
 public class XPackClientPlugin extends Plugin implements ActionPlugin, NetworkPlugin {
 
     @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
-    static Optional<String> X_PACK_FEATURE = Optional.of("x-pack");
+    public static Optional<String> X_PACK_FEATURE = Optional.of("x-pack");
```

Comment:
I don't think we should do this. I think we should add an interface `XPackPersistentTaskParams` that has a default implementation of `getRequiredFeature`. I think we want to do this so that we can later (not in this PR) add a check that all implementations of `PersistentTaskParams` in the X-Pack codebase extend `XPackPersistentTaskParams` and therefore are certain that they all return the correct value for `getRequiredFeature`. This would be a start:

```diff
diff --git a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java
index 602f4bdbc07..d91deb2d011 100644
--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java
@@ -40,6 +40,7 @@ import org.elasticsearch.license.LicenseService;
 import org.elasticsearch.license.LicensesMetaData;
 import org.elasticsearch.license.Licensing;
 import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.persistent.PersistentTaskParams;
 import org.elasticsearch.plugins.ExtensiblePlugin;
 import org.elasticsearch.plugins.ScriptPlugin;
 import org.elasticsearch.rest.RestController;
@@ -331,4 +332,13 @@ public class XPackPlugin extends XPackClientPlugin implements ScriptPlugin, Exte
 
     }
 
+    public interface XPackPersistentTaskParams extends PersistentTaskParams {
+
+        @Override
+        default Optional<String> getRequiredFeature() {
+            return XPackClientPlugin.X_PACK_FEATURE;
+        }
+
+    }
+
 }
diff --git a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/OpenJobAction.java b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/OpenJobAction.java
index 28b2889fcd3..115edc5b993 100644
--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/OpenJobAction.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/OpenJobAction.java
@@ -25,6 +25,7 @@ import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.persistent.PersistentTaskParams;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.xpack.core.XPackClientPlugin;
+import org.elasticsearch.xpack.core.XPackPlugin;
 import org.elasticsearch.xpack.core.ml.MachineLearningField;
 import org.elasticsearch.xpack.core.ml.job.config.Job;
 import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
@@ -129,7 +130,7 @@ public class OpenJobAction extends Action<OpenJobAction.Request, OpenJobAction.R
         }
     }
 
-    public static class JobParams implements PersistentTaskParams {
+    public static class JobParams implements XPackPlugin.XPackPersistentTaskParams {
 
         /** TODO Remove in 7.0.0 */
         public static final ParseField IGNORE_DOWNTIME = new ParseField("ignore_downtime");
@@ -245,10 +246,6 @@ public class OpenJobAction extends Action<OpenJobAction.Request, OpenJobAction.R
             return Version.V_5_4_0;
         }
 
-        @Override
-        public Optional<String> getRequiredFeature() {
-            return XPackClientPlugin.X_PACK_FEATURE;
-        }
     }
 
     public static class Response extends AcknowledgedResponse {
```

This this field can remain non-public.

Reply:
> a check that all implementations of PersistentTaskParams in the X-Pack codebase extend XPackPersistentTaskParams

I felt it was an overkill to introduce an interface, but if that's the direction I will follow and add it.

Reply:
I don't think it's overkill since it doesn't add any methods. It is more about providing a default implementation (so we are using it more like a poor man's mix-in) and also giving us a way to ensure via testing that every custom in the X-Pack code base returns that it requires the `x-pack` feature.